src/aero_reps/aero_rep_single_particle.c

```diff
@@ -187,6 +187,117 @@ void aero_rep_single_particle_get_effective_radius__m(
   return;
 }
 
+/** \brief Get the surface area of specified particle layer \f$r_{eff}\f$ (m)
+ *
+ * Solve for the surface area of the interfacial layer that exists between the 
+ * two phases considered in aerosol phase mass tranfer between layers. When more
+ * than one phase exists in a layer, a "fractional volume overlap" configuration 
+ * is applied (see CAMP Github Documentation for details).
+ * 
+ * \param model_data Pointer to the model data, including the state array
+ * \param aero_phase_idx_first Index of the first aerosol phase within the representation
+ * \param aero_phase_idx_second Index of the second aerosol phase within the representation
+ * \param surface_area Pointer to surface area of inner layer (m^2)
+ * \param partial_deriv \f$\frac{\partial r_{eff}}{\partial y}\f$ where \f$y\f$
+ *                      are species on the state array
+ * \param aero_rep_int_data Pointer to the aerosol representation integer data
+ * \param aero_rep_float_data Pointer to the aerosol representation
+ *                            floating-point data
+ * \param aero_rep_env_data Pointer to the aerosol representation
+ *                          environment-dependent parameters
+ * \param surf_area_layer Surface area of specified layer (m2)
+ */
+
+void aero_rep_single_particle_get_interface_surface_area__m2(
+    ModelData *model_data, int aero_phase_idx_first, int aero_phase_idx_second, 
+    double *surface_area, double *partial_deriv, 
+    int *aero_rep_int_data, double *aero_rep_float_data, double *aero_rep_env_data) {
+
+  int *int_data = aero_rep_int_data;
+  double *float_data = aero_rep_float_data;
+  double *curr_partial = NULL;
+  int layer_first = -1;
+  int layer_second = -1;
+  int layer_interface = -1;
+  int phase_model_data_id_first = -1;
+  int phase_model_data_id_second = -1;
+  double radius;
+  
+  int i_phase_count = 0;
+  for (int i_layer = 0; i_layer < NUM_LAYERS_; ++i_layer) {
+    for (int i_phase = 0; i_phase < NUM_PHASES_(i_layer); ++i_phase) {
+      if (LAYER_PHASE_START_(i_layer) <= aero_phase_idx_first && 
+          aero_phase_idx_first <= LAYER_PHASE_END_(i_layer) && 
+          i_phase_count == aero_phase_idx_first) {
+        layer_first = i_layer;
+        phase_model_data_id_first = PHASE_MODEL_DATA_ID_(i_layer, i_phase);
```

Comment:
If `phase_model_idx_first` and `phase_model_idx_second` can be figured out during initialization, could these be returned from the fortran function that returns pairs of indices for adjacent phases?